backend/Controllers/raceResults.js

```diff
@@ -130,9 +146,109 @@ router.get('/race-results', (req, res) => {
       .project({ 'calendar.drivers': 1 })
       .toArray()
       .then((results) => {
-        drivers = results;
+        drivers = results[0].calendar.drivers;
       });
-    return drivers[0];
+    return drivers;
+  };
+
+  const getTeamsList = async () => {
+    let teams;
+    await client
+      .db('RGL')
+      .collection('divisions')
+      .find({
+        division: req.query.division,
+      })
+      .project({ 'calendar.teams': 1 })
+      .toArray()
+      .then((results) => {
+        teams = results[0].calendar.teams;
+      });
+    return teams;
+  };
+
+  const setDriversDetails = async (results) => {
+    let driversList = await getDriversList();
+    let raceFormats = Object.keys(results.calendar.raceFormat);
+
+    for (const raceSession of raceFormats) {
+      let selectedSessionResults =
+        results.calendar.races[0].adjustedResults[raceSession];
+
+      for (const [index, driver] of selectedSessionResults.entries()) {
+        let selectedDriverNick = selectedSessionResults[index].nick;
+
+        driver.fullName = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).fullName;
+
+        driver.number = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).number;
+
+        driver.team = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).team;
+      }
+    }
+
+    return results.calendar.races[0].adjustedResults;
+  };
+
+  const setTeamsDetails = async (results) => {
+    let teamsList = await getTeamsList();
+    let raceFormats = Object.keys(results.calendar.raceFormat);
+
+    for (const raceSession of raceFormats) {
+      let selectedSessionResults =
+        results.calendar.races[0].adjustedResults[raceSession];
+
+      for (const [index, driver] of selectedSessionResults.entries()) {
+        let selectedDriverTeam = selectedSessionResults[index].team;
```

Comment:
To samo co wyżej

Reply:
Teraz pytanie: mogę wprowadzić tę zmianę, ale w takim wypadku musiałbym chwilę poczekać na wyniki z bazy. Musiałbym wprowadzić taką linijkę:
    let driversList = await getDriversList();

Co w takich okolicznościach będzie lepsze? Większa czytelność kodu, czy lepsza wydajność?

Reply:
przecież to już jest, linijki 171 oraz 199

Reply:
to samo tzn tutaj zrobić 
let selectedTeam=  teamsList.find(
          (team) => team.name === selectedDriverTeam
        )
chodziło o to, żeby przypisać do zmiennej wartość zamiast 3x .find()